components/Typography/tests/unit/MaterialScalableFontTests.m

```diff
@@ -0,0 +1,266 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTypography.h"
+
+#import "MaterialMath.h"
+#import "../../src/private/UIContentSizeCategory+Material.h"
+
+@interface UIFont_MaterialScalableTests : XCTestCase
+
+@end
+
+@implementation UIFont_MaterialScalableTests
+
+- (void)testScaledFontsReturnEquivalentFonts {
+  // Given
+  NSArray<MDCTextStyle> *textStyles = @[
+    MDCTextStyleHeadline1,
+    MDCTextStyleHeadline2,
+    MDCTextStyleHeadline3,
+    MDCTextStyleHeadline4,
+    MDCTextStyleHeadline5,
+    MDCTextStyleHeadline6,
+    MDCTextStyleSubtitle1,
+    MDCTextStyleSubtitle2,
+    MDCTextStyleBody1,
+    MDCTextStyleBody2,
+    MDCTextStyleButton,
+    MDCTextStyleCaption,
+    MDCTextStyleOverline,
+  ];
+
+  for (MDCTextStyle textStyle in textStyles) {
+    // When
+    UIFont *font1 = [UIFont systemFontOfSize:18.0];
```

Comment:
Tip: Consider using a different base font in different tests to make sure there's nothing accidentally special about "18".

Reply:
Done.